### PR TITLE
Update build scripts for Xcode 7.2

### PR DESCRIPTION
**Pull request by jmoody**
### Motivation

Resolves:

* Xcode 7.2: builds frank simulator library to Debug-iphoneos directory #302
* Xcode 7.2: builds simulator dylib to Debug-iphoneos directory #303

The problem is Xcode 7.2 requires:

```
EFFECTIVE_PLATFORM_NAME="-iphonesimulator"
```

### To Review

Reviewer must have Xcode 7.2

```
$ git fetch
$ git clone -t origin/feature/update-build-scripts-for-xcode-72
$ bundle install
$ make frank
$ make dylib
$ make app-cal
```

### CODE REVIEW

File: bin/make/frank-plugin.sh [hunk 1]
```diff
@@ -95,6 +101,7 @@ xcrun xcodebuild build \
   ARCHS="i386 x86_64" \
   VALID_ARCHS="i386 x86_64" \
   ONLY_ACTIVE_ARCH=NO \
+  EFFECTIVE_PLATFORM_NAME="-iphonesimulator" \
```
Comment (sapieneptus) [inline bin/make/frank-plugin.sh]: I don't really have enough context to ask good questions here, so please bear with me- but why is this line necessary ?

Comment (jmoody) [inline bin/make/frank-plugin.sh]: Starting in Xcode 7.2, simulator binaries (apps and libraries) are being built to `Debug-iphoneos` and not to `Debug-iphonesimulator`.   After exploring a bunch of other build settings, I found that this was the missing argument.

Comment (jmoody) [inline bin/make/frank-plugin.sh]: In case it was not clear, we could not build the dylibs or frank libraries with Xcode 7.2.

Comment (sapieneptus) [inline bin/make/frank-plugin.sh]: Got it - How did you figure out that this was the missing arg? Seems like a very non-obvious thing to have caused that result.

Comment (jmoody) [inline bin/make/frank-plugin.sh]: It was not the first thing I tried.

I started by going through the Xcode build logs from Xcode looking at what variables were available to xcodebuild.  I came up with this brute force approach which forced Xcode to build to very specific directories.


```
OBJECT_ROOT_DIR="${XC_BUILD_DIR}/Build/Intermediates/${XC_CONFIG}-iphonesimulator"

xcodebuild
    ...
    OBJROOT="${OBJECT_ROOT_DIR}" \
    SHARED_DERIVED_FILE_DIR="${BUILD_PRODUCTS_DIR}" \
    CONFIGURATION_BUILD_DIR="${BUILD_PRODUCTS_DIR}" \
    BUILT_PRODUCTS_DIR="${BUILD_PRODUCTS_DIR}" \
    TARGET_BUILD_DIR="${BUILD_PRODUCTS_DIR}" \
    DWARF_DSYM_FOLDER_PATH="${BUILD_PRODUCTS_DIR}" \
```

This works, but it is a lot of effort.  I noticed the `EFFECTIVE_PLATFORM_NAME` variable early on but thought, "That can't be the problem...".   Turns out that it was. :8ball:
